server/lib/src/idm/ldap.rs

```diff
@@ -642,16 +645,35 @@ mod tests {
 
         let pce = UnixPasswordChangeEvent::new_internal(UUID_ADMIN, TEST_PASSWORD);
 
-        assert!(idms_prox_write.set_unix_account_password(&pce).is_ok());
+        let allow_unix_pw_flag = ModifyEvent::new_internal_invalid(
+            filter!(f_eq(Attribute::Uuid, PartialValue::Uuid(UUID_DOMAIN_INFO))),
+            ModifyList::new_purge_and_set(Attribute::DomainLdapAllowUnixPwBind, Value::Bool(false)),
+        );
+        assert!(idms_prox_write.qs_write.modify(&allow_unix_pw_flag).is_ok());
         assert!(idms_prox_write.commit().is_ok());
-
         let anon_t = ldaps.do_bind(idms, "", "").await.unwrap().unwrap();
         assert!(anon_t.effective_session == LdapSession::UnixBind(UUID_ANONYMOUS));
         assert!(
             ldaps.do_bind(idms, "", "test").await.unwrap_err() == OperationError::NotAuthenticated
         );
 
         // Now test the admin and various DN's
+
+
+        // Writing test for optional use of POSIX password for bind
+        // NOT ALLOWING UNIX_PW FOR BIND :
+        let admin_t = ldaps.do_bind(idms, "admin", TEST_PASSWORD).await.unwrap(); // Cannot unwrap futher as on disallowed unix_pw_bind it will return Ok(None)
+        assert!(admin_t.is_none() == true);
+
+        // ALLOWING UNIX_PW FOR BIND :
+        let mut idms_prox_write = idms.proxy_write(duration_from_epoch_now()).await;
```

Comment:
Shouldn't you be setting this to "false" to show what happens if you make it false?

Reply:
I think your logic here is fliipped btw. when the value is *absent* unix bind should work.
when allow pw bind is true it should work
and when it's false, it should *not* work.

So because you're never setting false, it isn't working I think. I think you need to more clearly seperate out the three cases in the test to match the states that I mentioned (about possibility of abscence, then the two possible bool states).